### ISSUE TITLE
Converting JSON -> YAML -> JSON errors if timestamp fields are involved

### ISSUE THREAD

**Issue by nicjohnson145** [TomWright/dasel#331]
**Describe the bug**

When converting from JSON to YAML and back again, fields that contain timestamps result in parse failures and errors

**To Reproduce**

```
❯ cat foo.json
{
    "createdAt": "2023-06-13T20:19:48.531620935Z"
}% 

❯ cat foo.json | dasel -r json -w yaml | dasel -r yaml -w json
Error: could not unmarshal data: unhandled scalar node tag: !!timestamp
```

However, if you write to yaml, quote the value, and then turn to json, it works just fine

```
❯ cat foo.json | dasel -r json -w yaml > foo.yaml

❯ cat foo.yaml
createdAt: 2023-06-13T20:19:48.531620935Z

❯ vim foo.yaml
                                                                                                                                                                                                                                                                                                                           
❯ cat foo.yaml
createdAt: "2023-06-13T20:19:48.531620935Z"
                                                                                                                                                                                                                                                                                                                           
❯ cat foo.yaml | dasel -r yaml -w json
{
  "createdAt": "2023-06-13T20:19:48.531620935Z"
}
```

**Expected behavior**

Converting from JSON to YAML and back should result in the original JSON (or equivalent JSON)


**Desktop (please complete the following information):**

- OS: Ubuntu 22.04
- Version: 2.3.4

```
❯ uname -a
Linux hermes 5.19.0-43-generic #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
                                                                                                                                                                                                                                                                                                                           
❯ dasel --version
dasel version v2.3.4
```

**Comment by TomWright** [comment]
This has been fixed and will be included in the next release.
```
$ echo '{"createdAt": "2023-06-13T20:19:48.531620935Z"}' | go dasel -r json -w yaml | dasel -r yaml -w json
{
  "createdAt": "2023-06-13T20:19:48.531620935Z"
}
```

**Comment by TomWright** [comment]
The release has been merged in homebrew and you can now upgrade to v2.3.5
